telemetry/telemetry_test.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"os"
+	"testing"
+	"context"
+	"time"
+	"sync"
+	"syscall"
+)
+
+func TestSignalHandler(t *testing.T) {
```

Comment:
This test can cover SignalHandler, how do we test main?

Reply:
Main function now does not contain any real functional logic besides calling setup flags and goroutines